RELEASE.md

```diff
@@ -51,9 +51,6 @@
 - [ ] _origin-js_: [Publish to npm](https://docs.npmjs.com/cli/publish). 
   - `npm publish`
 ### demo-dapp
-- [ ] _demo-dapp_: Replace git reference with version number in `package.json`
-  -   ` "origin": "OriginProtocol/origin-js#develop"` --> `"origin": "0.6.1"`
```

Comment:
@wanderingstan even though we don't need to change this from GitHub to the version number, do we still need to update the version number manually?